java-components/cli/src/main/java/io/github/redhatappstudio/jvmbuild/cli/builds/BuildLogsCommand.java

```diff
@@ -90,88 +109,151 @@ public void run() {
-                        int r;
-                        byte[] buff = new byte[1024];
-                        while ((r = in.read(buff)) > 0) {
-                            System.out.print(new String(buff, 0, r));
+                    List<ContainerStatus> containerStatuses = new ArrayList<>(pod.get().getStatus().getContainerStatuses());
+                    containerStatuses.sort(Comparator.comparing(
+                            t -> OffsetDateTime.parse(t.getState().getTerminated().getFinishedAt(), formatter)));
+                    for (var i : containerStatuses) {
+                        var p = pod.inContainer(i.getName());
+
+                        System.out.println("### Logs for container " + i.getName());
+                        System.out.println("#####################################################\n\n");
+                        try (var w = p.watchLog(); var in = w.getOutput()) {
+                            int r;
+                            byte[] buff = new byte[1024];
+                            while ((r = in.read(buff)) > 0) {
+                                System.out.print(new String(buff, 0, r));
+                            }
+                        } catch (IOException e) {
+                            throw new RuntimeException(e);
                         }
-                    } catch (IOException e) {
-                        throw new RuntimeException(e);
+
+                        var finishTime = OffsetDateTime.parse(i.getState().getTerminated().getFinishedAt(), formatter);
+                        Duration duration = Duration.between(startTime, finishTime);
+                        startTime = finishTime;
+                        System.out.println("\n\n#####################################################");
+                        System.out.println(
+                                "### Container " + i.getName() + " finished at " + i.getState().getTerminated().getFinishedAt()
+                                        + " in " + duration.getSeconds() + " seconds");
                     }
+                }
+                System.out.println("#####################################################\n\n");
+            }
+
+        } else {
 
-                    var finishTime = OffsetDateTime.parse(i.getState().getTerminated().getFinishedAt(), formatter);
-                    Duration duration = Duration.between(startTime, finishTime);
-                    startTime = finishTime;
-                    System.out.println("\n\n#####################################################");
-                    System.out.println(
-                            "### Container " + i.getName() + " finished at " + i.getState().getTerminated().getFinishedAt()
-                                    + " in " + duration.getSeconds() + " seconds");
+            RouteSpec routeSpec = client.routes().inNamespace("openshift-pipelines").withName("tekton-results").get().getSpec();
```

Comment:
Rather than having to have another CLI option to pass in the route, I thought we could just look up the one defined by `tekton_route.yaml`.

Reply:
This is a good idea in theory, however for RHTAP the users don't have any permissions on the openshift-pipelines namespace.

Can we do some sensible defaults based on the cluster address?

For RHTAP the results path looks like:

`https://console.redhat.com/api/k8s/plugins/tekton-results/workspaces/sdouglas1/apis/results.tekton.dev`

(this is due to it going through a proxy)

Maybe attempt to lookup the address, if it fails (likely due to permissions), default to the RHTAP prod one?

Reply:
I've changed to to attempt the routes call and if that fails fall back to assembling a prod url. I've made the 'prod' url customisable through the CLI in case someone is using staging.